### PR TITLE
AAP-51940 [devel-only] Remove role-tracking feature no longer used

### LINKED ISSUE #159: [RBAC] Remove the "relationship tracking" or "role tracking" feature
We don't need this anymore. JWT data exchange details were unclear at the time.

Tests at `test_app/tests/rbac/features/test_role_tracking.py`

### PR DESCRIPTION
**Pull request by AlanCoding**
## Description
Fixes https://github.com/ansible/django-ansible-base/issues/159

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [x] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist
- [x] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

---

> [!NOTE]
> Removes the tracked relationship feature, its registry/hooks, associated M2M fields/usages, and corresponding documentation/tests.
> 
> - **RBAC core**:
>   - Remove tracked relationship syncing: delete `TrackedRelationship`, `permission_registry.track_relationship`, and tracker initialization/state.
>   - Drop sync calls from `RoleDefinition.give_or_remove_permission`.
> - **Models/Migrations (test_app)**:
>   - Remove Organization `users/admins`, Team `users/admins/team_parents` M2M fields; adjust migrations accordingly.
> - **Docs**:
>   - Remove the "Tracked Relationships" section from `docs/apps/rbac/for_app_developers.md`.
> - **Tests**:
>   - Delete tests relying on tracked relationships and related user sublists; trim unrelated helper/tests accordingly.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 28bdb85f3e8f8cbc2de5a8aec89f0b93ae710724. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: test_app/migrations/0006_team_admins_team_users.py [hunk 1]
```diff
@@ -11,14 +10,5 @@ class Migration(migrations.Migration):
...
-            model_name='team',
-            name='admins',
-            field=models.ManyToManyField(blank=True, help_text='The list of admins for this team', related_name='teams_administered', to=settings.AUTH_USER_MODEL),
-        ),
-        migrations.AddField(
-            model_name='team',
-            name='users',
-            field=models.ManyToManyField(blank=True, help_text='The list of users on this team', related_name='teams', to=settings.AUTH_USER_MODEL),
-        ),
+        # Removed team admins and user relationsips
```
[the remark this answers is not in the document]
Comment (AlanCoding) [reply test_app/migrations/0006_team_admins_team_users.py]: I am fully aware of this, but I refer to the name "test_app". It would be an absurd and infuriating requirement to support migrations in our app for testing DAB. Hard-no.

**Review by fosterseth — APPROVED** [review]
